system/shruntime.py

```diff
@@ -476,7 +487,13 @@ def exec_py_file(self, filename,
 
         saved_sys_argv = sys.argv[:]
         # First argument is the script name
-        sys.argv = [os.path.basename(filename)] + (args or [])
+        argv = [os.path.basename(filename)] + (args or [])
+        
+        # convert sys.argv to unicode if on python3
+        if PY3:
+        	# todo: this is only a temporary solution and needs to be redone. This may be buggy.
+        	argv = [c if isinstance(c, unicode) else c.decode("latin-1") for c in argv]
```

Comment:
`"latin-1"` is not a good default encoding, it only works for a very small number of characters and is almost never used anymore. `"utf-8"` is the de facto standard, that would be a better choice. (Or simply leave out the encoding completely - Python defaults to `"utf-8"` if you don't pass an encoding to `decode`/`encode`.)

Reply:
+1 on utf-8

Reply:
IIRC `"latin-1"` has better bytes compatibility. Decoding a random bytestring (e.g. `os.urandom(128)`) in `"utf-8"` will likely raise an Error, but `"latin-1"` can handle it.
Anyway, this was only meant as a quick fix and the encoding should probably be moved to to the `stash.system.*` module responsible for the input.